var/spack/repos/builtin/packages/python/package.py

```diff
@@ -431,6 +448,83 @@ def flag_handler(self, name, flags):
         # allow flags to be passed through compiler wrapper
         return (flags, None, None)
 
+    @property
+    def plat_arch(self):
+        """
+        String referencing platform architecture
+        filtered through Python's Windows build file
+        architecture support map
+
+        Note: This function really only makes
+        sense to use on Windows, could be overridden to
+        cross compile however.
+        """
+        arch = platform.machine()
+        if arch in arch_map:
```

Comment:
It in fact very well may. This filter was intended to capture a few known values and convert them into what Python's batch script expects. It could be that an architecture actually returns one of the desired values already, or that it returns an architecture that would be appropriately handled by Python's handling of unexpected architectures (defaulting to Win32). Either way, we could certainly warn here, I have no issue with that, but I will say that Python's build system and batch script report architectural inconsistencies fairly well IMO. The only issue would be where we want 64bit and that Python will be building in 32 as default. That build would still succeed, so Python would not report.